tests/perf/actor_id_scale/actor_id_scale_test.go

```diff
@@ -0,0 +1,107 @@
+//go:build perf
+// +build perf
+
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actor_id_scale
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/dapr/dapr/tests/perf/utils"
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/dapr/dapr/tests/runner/loadtest"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	numHealthChecks        = 60 // Number of times to check for endpoint health per app.
+	serviceApplicationName = "perf-actorfeatures"
+	actorType              = "scale-id"
+)
+
+var (
+	tr *runner.TestRunner
+)
+
+func TestMain(m *testing.M) {
+	utils.SetupLogs("actor_id_stress_test")
+
+	testApps := []kube.AppDescription{
+		{
+			AppName:           serviceApplicationName,
+			DaprEnabled:       true,
+			ImageName:         "perf-actorfeatures",
+			Replicas:          10,
+			IngressEnabled:    true,
+			MetricsEnabled:    true,
+			AppPort:           3000,
+			DaprCPULimit:      "4.0",
+			DaprCPURequest:    "0.1",
+			DaprMemoryLimit:   "512Mi",
+			DaprMemoryRequest: "250Mi",
+			AppCPULimit:       "4.0",
+			AppCPURequest:     "0.1",
+			AppMemoryLimit:    "512Mi",
+			AppMemoryRequest:  "256Mi",
+			AppEnv: map[string]string{
+				"TEST_APP_ACTOR_TYPE": actorType,
+			},
+		},
+	}
+
+	tr = runner.NewTestRunner("actoridstress", testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+func TestActorIdStress(t *testing.T) {
+	// Get the ingress external url of test app
+	testServiceAppURL := tr.Platform.AcquireAppExternalURL(serviceApplicationName)
+	require.NotEmpty(t, testServiceAppURL, "test app external URL must not be empty")
```

Comment:
TIL about require! I was baffled that `assert` would just proceed with test execution and that was "normal". Thanks.